R/paste_skip_na.R

```diff
@@ -0,0 +1,64 @@
+#' Concatenate strings dropping missing values
+#'
+#' @details If all values are missing, the value from the first argument is
+#'   preserved.
+#'
+#' @param ...,sep,collapse See \code{?paste}
+#' @return A character vector of pasted values.
+#' @examples
+#' paste_skip_na(NA) # NA_character_
+#' paste_skip_na("A", NA) # "A"
+#' paste_skip_na("A", NA, c(NA, "B"), sep = ",") # c("A", "A,B")
+#' @export
+paste_skip_na <- function(..., sep=" ", collapse=NULL) {
+  args <- list(...)
+  if (length(args) <= 1) {
+    if (length(args) == 0) {
+      # match the behavior of paste
+      paste(sep=sep, collapse=collapse)
+    } else if (!is.null(collapse)) {
+      if (all(is.na(args[[1]]))) {
+        # Collapsing with all NA values results in NA
+        NA_character_
+      } else {
+        # Collapsing without all NA values collapses the non-NA values
+        paste(na.omit(args[[1]]), sep=sep, collapse=collapse)
+      }
+    } else {
+      # as.character() to ensure that logical NA values are converted to
```

Comment:
`paste_skip_na(c("A", "B", "C", NA))` returns the input.  That is better than `paste()` which quotes the `NA` -- but as a naive user I assumed this would return `c("A", "B", "C")`.  I think I'm missing something but what?

Reply:
I have a strong preference that the input length is the same as the output length.  That way, the output can be used as part of a tidy pipeline.  The following code requires the same length input and output:

```
data.frame(A = c("B", NA)) %>% mutate(C = paste(A))
```

Reply:
Ah of course, I was only thinking about single values and not vectorization on a column.  100% agree.